### PR TITLE
[SYCL] Avoid spurious error about call through function pointer

### PR DESCRIPTION
**Pull request by premanandrao**
Signed-off-by: Premanand M Rao <premanand.m.rao@intel.com>

### CODE REVIEW

File: clang/test/SemaSYCL/sycl-fptr-lambda.cpp [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+template <typename name, typename Func>
+__attribute__((sycl_kernel)) void kernel_single_task(Func kernelFunc) {
+  kernelFunc();
+}
+
+int main() {
+  kernel_single_task<class fake_kernel>([]() {
+		                   // expected-no-error@+1
+                                   auto l = [](auto f){f();};
+				             });
```
Comment (bader) [inline clang/test/SemaSYCL/sycl-fptr-lambda.cpp]: Is this file `clang-format`-ed?

File: clang/test/SemaSYCL/sycl-fptr-lambda.cpp [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+// expected-no-diagnostics
+
+template <typename name, typename Func>
+__attribute__((sycl_kernel)) void kernel_single_task(Func kernelFunc) {
+  kernelFunc();
+}
+
+int main() {
+  kernel_single_task<class fake_kernel>([]() {
+		                   // expected-no-error@+1
```
Comment (bader) [inline clang/test/SemaSYCL/sycl-fptr-lambda.cpp]: We already have `// expected-no-diagnostics` at the beginning of the file, so this check seems redundant. Right?

File: clang/test/SemaSYCL/sycl-fptr-lambda.cpp (hunk 1 above)
Comment (premanandrao) [reply clang/test/SemaSYCL/sycl-fptr-lambda.cpp]: Sorry, no, now fixed.

File: clang/test/SemaSYCL/sycl-fptr-lambda.cpp (hunk 2 above)
Comment (premanandrao) [reply clang/test/SemaSYCL/sycl-fptr-lambda.cpp]: Yes, it is.  Removed.  Thanks.

**Comment by premanandrao** [comment]
@bader, please merge this change when you find time.   Thanks!

**Pull request merged**